src/modules/achievementSchemas.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from 'zod';
+
+import { tr } from './modules.i18n';
+
+export const createAndGiveAchievementSchema = z.object({
+    icon: z.string({ required_error: tr('Icon URL is required') }).min(1, { message: tr('Icon URL is required') }),
+    title: z
+        .string({ required_error: tr('Title is required') })
+        .min(3, { message: tr('Title must be longer than {min} symbol', { min: 3 }) }),
+    description: z
+        .string({ required_error: tr('Description is required') })
+        .min(3, { message: tr('Description must be longer than {min} symbol', { min: 3 }) }),
+    userId: z.string(),
+    userEmail: z.string(),
```

Comment:
Why email is sent from frontend?

Reply:
to send email to user without getting user email from db: we already have it

Reply:
It's not safe, i can trick server to send email to any address